<launch>
    <!-- 参数 -->
    <arg name="llm_api_url" default="http://localhost:8000/generate" />
    <arg name="api_key" default="" />
    <arg name="map_width" default="10" />
    <arg name="map_height" default="10" />
    
    <!-- 激光雷达障碍物探测器 -->
    <node name="lidar_obstacle_detector" pkg="llm_navigation" type="lidar_obstacle_detector.py" output="screen">
        <param name="distance_threshold" value="2.0" />
        <param name="cluster_threshold" value="0.3" />
        <param name="min_points_per_obstacle" value="3" />
    </node>
    
    <!-- LLM路径规划器 -->
    <node name="llm_path_planner" pkg="llm_navigation" type="llm_path_planner.py" output="screen">
        <param name="llm_api_url" value="$(arg llm_api_url)" />
        <param name="api_key" value="$(arg api_key)" />
        <param name="map_width" value="$(arg map_width)" />
        <param name="map_height" value="$(arg map_height)" />
        <param name="grid_resolution" value="0.1" />
    </node>
    
    <!-- 路径跟随器 -->
    <node name="path_follower" pkg="llm_navigation" type="path_follower.py" output="screen">
        <param name="waypoint_reach_distance" value="0.2" />
        <param name="linear_velocity" value="0.3" />
        <param name="angular_velocity" value="0.5" />
        <param name="robot_frame" value="base_link" />
    </node>
    
    <!-- RViz可视化 -->
    <node name="rviz" pkg="rviz" type="rviz" args="-d $(find llm_navigation)/rviz/navigation.rviz" />
</launch>
